### PR TITLE
Run tests for showcases

### PR DESCRIPTION
**Pull request by pbt-santos**
Adds workflow to test showcases daily, and on PR or commit on master.

Showcase tests consist of a roundtrip compilation test with latest legend versions to enforce compilation and formatting. Does not yet allow for comments.

The paths in the poms and the test file are different due to the location that workflows vs tests checkout the code.

Showcase code moved to not clog up directory structure.

### CODE REVIEW

File: .github/workflows/showcase-test.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: showcase-tests
+run-name: Showcase Tests
+on:
```
Comment (akphi) [inline .github/workflows/showcase-test.yml:3]: How about this?

```yaml
  schedule:
    - cron: '0 0,12 * * *' # run twice daily
  push:
    branches:
      - master
    paths:
      - 'showcases/'
  pull_request:
    branches:
      - '**'
    paths:
      - 'showcases/'
```

File: .github/workflows/showcase-test.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup Java JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'oracle'
+          cache: 'maven'
```
Comment (akphi) [inline .github/workflows/showcase-test.yml:25]: will this be equivalent to using `actions/cache` like in https://github.com/finos/legend-engine/blob/master/.github/workflows/build.yml ?

File: showcases/pom.xml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    <groupId>org.finos</groupId>
+    <artifactId>legend-showcases</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <name>Legend Showcases</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>1.7</maven.compiler.source>
```
Comment (akphi) [inline showcases/pom.xml:15]: why `1.7`?

File: showcases/pom.xml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            <artifactId>junit</artifactId>
+            <version>4.13.2</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <pluginManagement><!-- lock down plugins versions to avoid using Maven defaults (may be moved to parent pom) -->
+            <plugins>
+                <!-- Fetch Snapshot versions -->
```
Comment (akphi) [inline showcases/pom.xml:47]: did you check that we need to specify all of these?

File: .gitignore [hunk 5]
```diff
@@ -34,3 +35,6 @@ docs/roadmap.md
 
 # Ignore checked out models
 docs/models
+
+# Ignore Java generated files
+target/
```
Comment (akphi) [inline .gitignore:40]:
```suggestion
showcases/target/
```

**Review by akphi — CHANGES_REQUESTED** [review]

File: .github/workflows/showcase-test.yml (hunk 2 above)
Comment (pbt-santos) [reply .github/workflows/showcase-test.yml:25]: setup-java uses actions/cache under the hood: https://github.com/actions/setup-java#caching-packages-dependencies

Looking at that file, I believe the simple usage should be equivalent.

**Review by akphi — APPROVED** [review]

**Pull request merged**